src/remat.jl

```diff
@@ -133,7 +133,15 @@ function getθ!(v::AbstractVector{T}, A::ReMat{T}) where {T}
     v
 end
 
-levels(A::ReMat) = A.levels
+function DataAPI.levels(A::ReMat)
+    pool = A.levels
+    present = falses(size(pool))
+    @inbounds for i in A.refs
+        present[i] = true
+        all(present) && return pool
```

Comment:
this short-circuits the `for` at the cost of iterating over `present` a lot via `all`. It might be easier to check whether `length(A.levels) == length(A.refs)` and short-circuit before the loop

Reply:
Not sure I understand.  `length(A.levels) == length(A.refs)` doesn't really tell you anything.  `length(A.refs)` is always the number of observations.  The only interesting cases for categorical data are when `length(A.levels)` is less than `length(A.refs)`, otherwise there is no compression in the representation.

My motive in doing the short-circuit check is that in the common cases, where `length(A.levels)` is small, `all(present)` would be a fast operation on this, comparatively small, `BitArray`, which is what `falses` produces.  If the number of levels does not exceed the word size this check can be done by masking in a register.  It takes a few nanoseconds on my machine.

I suspect that the short circuit will never make a noticeable difference so we can remove it if doing so will make the code clearer.

Reply:
Ah, okay, that was a thinko on my part. I think this will be fine given the typical  nlevels << nobs.